pandas/tests/tools/test_to_datetime.py

```diff
@@ -3587,3 +3672,29 @@ def test_from_numeric_arrow_dtype(any_numeric_ea_dtype):
     result = to_datetime(ser)
     expected = Series([1, 2], dtype="datetime64[ns]")
     tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "date, date_expected, warning",
+    [
+        (["2020-01-01 00:00+00:00", "2020-01-01 00:00+02:00", ""], None, FutureWarning),
+        (
+            ["2020-01-01 00:00+00:00", ""],
+            [Timestamp("2020-01-01 00:00+00:00"), "NaT"],
+            None,
+        ),
+    ],
+)
+def test_to_datetime_with_empty_str_utc_false_format_mixed(
+    date, date_expected, warning
+):
+    # GH 50887
+    msg = "parsing datetimes with mixed time zones will raise a warning"
+
+    if warning is not None:
+        with tm.assert_produces_warning(warning, match=msg):
+            to_datetime(date, format="mixed")
+    else:
```

Comment:
this looks quite complicated, esp. for just two test cases. can we split it into two separate tests please?

Reply:
sure, I made two tests.